external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/MsSqlServerIntegrationSuite.scala

```diff
@@ -0,0 +1,173 @@
+    assert(rows(0).getDouble(4) == 1.23456788103168E14)
+    assert(rows(0).getLong(5) == 9223372036854775807L)
+    assert(rows(0).getInt(6) == 2147483647)
+    assert(rows(0).getInt(7) == 32767)
+    assert(rows(0).getInt(8) == 255)
+    assert(rows(0).getAs[BigDecimal](9).equals(new BigDecimal("922337203685477.5800")))
+    assert(rows(0).getAs[BigDecimal](10).equals(new BigDecimal("214748.3647")))
+  }
+
+  test("Date types") {
+    val df = sqlContext.read.jdbc(jdbcUrl, "dates", new Properties)
+    val rows = df.collect()
+    assert(rows.length == 1)
+    val types = rows(0).toSeq.map(x => x.getClass.toString)
+    assert(types.length == 6)
+    assert(types(0).equals("class java.sql.Date"))
+    assert(types(1).equals("class java.sql.Timestamp"))
+    assert(types(2).equals("class java.sql.Timestamp"))
+    assert(types(3).equals("class java.lang.String"))
+    assert(types(4).equals("class java.sql.Timestamp"))
+    assert(types(5).equals("class java.sql.Timestamp"))
+    assert(rows(0).getAs[Date](0).equals(Date.valueOf("1991-11-09")))
+    assert(rows(0).getAs[Timestamp](1).equals(Timestamp.valueOf("1999-01-01 13:23:35.0")))
+    assert(rows(0).getAs[Timestamp](2).equals(Timestamp.valueOf("9999-12-31 23:59:59.0")))
+    assert(rows(0).getString(3).equals("1901-05-09 23:59:59.0000000 +14:00"))
+    assert(rows(0).getAs[Timestamp](4).equals(Timestamp.valueOf("1996-01-01 23:24:00.0")))
+    assert(rows(0).getAs[Timestamp](5).equals(Timestamp.valueOf("1900-01-01 13:31:24.0")))
+  }
+
+
+  test("String types") {
+    val df = sqlContext.read.jdbc(jdbcUrl, "strings", new Properties)
+    val rows = df.collect()
+    assert(rows.length == 1)
+    val types = rows(0).toSeq.map(x => x.getClass.toString)
+    assert(types.length == 9)
+    assert(types(0).equals("class java.lang.String"))
+    assert(types(1).equals("class java.lang.String"))
+    assert(types(2).equals("class java.lang.String"))
+    assert(types(3).equals("class java.lang.String"))
+    assert(types(4).equals("class [B"))
+    assert(types(5).equals("class [B"))
+    assert(types(6).equals("class java.lang.String"))
+    assert(types(7).equals("class java.lang.String"))
+    assert(types(8).equals("class [B"))
+    assert(rows(0).getString(0).length == 10)
+    assert(rows(0).getString(0).trim.equals("the"))
+    assert(rows(0).getString(1).equals("quick"))
+    assert(rows(0).getString(2).length == 10)
+    assert(rows(0).getString(2).trim.equals("brown"))
+    assert(rows(0).getString(3).equals("fox"))
+    assert(java.util.Arrays.equals(rows(0).getAs[Array[Byte]](4), Array[Byte](0, 1, -30, 64)))
+    assert(java.util.Arrays.equals(rows(0).getAs[Array[Byte]](5), Array[Byte](0, 1, -30, 64)))
+    assert(rows(0).getString(6).equals("the"))
+    assert(rows(0).getString(7).equals("lazy"))
+    assert(java.util.Arrays.equals(rows(0).getAs[Array[Byte]](8), Array[Byte](100, 111, 103)))
+  }
+
+
+  test("Basic write test") {
```

Comment:
@lipzhu . Does this test case pass on your environment? For me, this fails with
- `mcr.microsoft.com/mssql/server:latest`
- `mcr.microsoft.com/mssql/server:2017-CU12-ubuntu`